main.go

```diff
@@ -1439,6 +1439,22 @@ func recordedAsNotObfuscated(obj types.Object) bool {
 	return ok
 }
 
+// These  are copied from astutil
+type visitFn func(node ast.Node)
+
+func (fn visitFn) Visit(node ast.Node) ast.Visitor {
+	fn(node)
+	return fn
+}
+
+// isTopName returns true if n is a top-level unresolved identifier with the given name.
+func isTopName(n ast.Expr, name string) bool {
+	id, ok := n.(*ast.Ident)
+	return ok && id.Name == name && id.Obj == nil
```

Comment:
I would really not rely on `go/ast`'s object tracking, because it doesn't work well at all. We already have proper type information via `go/types`, so you should use that instead.